File: kattis/busassignment/busassignment.py
import sys
 
class busassignment:
 
    def __init__(self, debug: bool = False):
        self.debug = debug
 
    def dprint(self, *args, **kwargs):
        if self.debug:
            print(*args, **kwargs)
 
    def findMaximumCapacity(self):
        # Find the number of stopage
        self.dprint("Input number of stopage.")
        inputNumberOfStopage = sys.stdin.readline()
        if not inputNumberOfStopage:
            return
 
        numberOfStopage = int(inputNumberOfStopage.strip())
        self.dprint(f"Number of stopage: {numberOfStopage}")
        self.dprint("-------------------------------------")
 
        # Process all passengers at each stopage
        currentPassenger = 0
        maximumCapacity = 0

        for i in range(numberOfStopage):
            # Find the number of passengers
            self.dprint(f"Input the off/on passengers for stopage {i+1}")
            inputNumberOfPassengers = sys.stdin.readline()
            if not inputNumberOfPassengers:
                break
 
            numberOfPassengers = inputNumberOfPassengers.strip().split()
            if len(numberOfPassengers) != 2:
                break
            getOffPassengers = int(numberOfPassengers[0])
            getOnPassengers = int(numberOfPassengers[1])
            self.dprint(f"Get off passengers: {getOffPassengers} and get on passengers: {getOnPassengers}")
 
            # Calculate net passengers at this stopage
            currentPassenger += (getOnPassengers - getOffPassengers)
            self.dprint(f"Current passengers: {currentPassenger}")
 
            # Calculate maximum capcity of the bus
            maximumCapacity = max(maximumCapacity, currentPassenger)
            self.dprint(f"Maximum capacity at stopage {i+1}: {maximumCapacity}")
            self.dprint("-------------------------------------")

        print(maximumCapacity)
        self.dprint("-------------------------------------")
 
if __name__ == "__main__":
    busassignment(False).findMaximumCapacity()

File: kattis/busassignment/busassignment.java
package kattis.busassignment;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class busassignment {

    private final boolean debug;
    private final BufferedReader bufferedReader;
    private StringTokenizer stringTokenizer;

    public busassignment(boolean debug) {
        this.debug = debug;
        this.bufferedReader = new BufferedReader(new InputStreamReader(System.in));
    }

    private void dprint(Object... args) {
        if (debug) {
            StringBuilder sb = new StringBuilder();
            for (Object arg : args) {
                sb.append(arg).append(" ");
            }
            System.out.println(sb.toString().trim());
        }
    }

    private String nextToken() throws IOException {
        while (stringTokenizer == null || !stringTokenizer.hasMoreTokens()) {
            String line = bufferedReader.readLine();
            if (line == null)
                return null;
            stringTokenizer = new StringTokenizer(line);
        }
        return stringTokenizer.nextToken();
    }

    public void findMaximumCapacity() {
        try {
            // Find the number of stoppage
            dprint("Input number of stoppage.");
            String inputNumberOfStoppage = nextToken();
            if (inputNumberOfStoppage == null)
                return;

            int numberOfStoppage = Integer.parseInt(inputNumberOfStoppage);
            dprint("Number of stoppage:", numberOfStoppage);
            dprint("-------------------------------------");

            // Process all passengers at each stoppage
            int currentPassenger = 0;
            int maximumCapacity = 0;

            for (int i = 0; i < numberOfStoppage; i++) {
                dprint("Input the off/on passengers for stoppage " + (i + 1));
                String offStr = nextToken();
                String onStr = nextToken();
                if (offStr == null || onStr == null)
                    break;

                int getOffPassengers = Integer.parseInt(offStr);
                int getOnPassengers = Integer.parseInt(onStr);
                dprint("Get off passengers:", getOffPassengers, "and get on passengers:", getOnPassengers);

                // Calculate net passengers at this stoppage
                currentPassenger += (getOnPassengers - getOffPassengers);
                dprint("Current passengers:", currentPassenger);

                // Calculate maximum capacity of the bus
                maximumCapacity = Math.max(maximumCapacity, currentPassenger);
                dprint("Maximum capacity at stoppage " + (i + 1) + ":", maximumCapacity);
                dprint("-------------------------------------");
            }

            // The actual output that prints regardless of the debug mode status
            System.out.println(maximumCapacity);
            dprint("-------------------------------------");
        } catch (Exception exception) {
            if (debug)
                exception.printStackTrace();
        }
    }

    public static void main(String[] args) {
        new busassignment(false).findMaximumCapacity();
    }
}